diesel/src/expression/sql_literal.rs

```diff
@@ -20,12 +21,52 @@ pub struct SqlLiteral<ST> {
 }
 
 impl<ST> SqlLiteral<ST> {
+    #[doc(hidden)]
     pub fn new(sql: String) -> Self {
         SqlLiteral {
             sql: sql,
             _marker: PhantomData,
         }
     }
+
+    /// Bind a value for use with this SQL query. This function should be used
+    /// with care, as Diesel cannot validate that the value is of the right type
+    /// nor can it validate that you have passed the correct number of
+    /// parameters.
+    ///
+    /// # Example
+    ///
+    /// ```rust
+    /// # #[macro_use] extern crate diesel;
+    /// # include!("src/doctest_setup.rs");
+    /// #
+    /// # table! {
+    /// #     users {
+    /// #         id -> Integer,
+    /// #         name -> VarChar,
+    /// #     }
+    /// # }
+    /// #
+    /// # fn main() {
+    /// #     use self::users::dsl::*;
+    /// #     use diesel::expression::dsl::sql;
+    /// #     use diesel::types::{Integer, Text};
+    /// #     let connection = establish_connection();
+    /// #[cfg(feature="postgres")]
+    /// let query = sql::<Integer>("SELECT id FROM users WHERE name = $1");
+    /// #[cfg(not(feature="postgres"))]
+    /// let query = sql::<Integer>("SELECT id FROM users WHERE name = ?");
+    /// let seans_id = query.clone().bind::<Text, _>("Sean")
+    ///     .get_result(&connection);
+    /// assert_eq!(Ok(1), seans_id);
+    /// let tess_id = query.bind::<Text, _>("Tess")
+    ///     .get_result(&connection);
+    /// assert_eq!(Ok(2), tess_id);
+    /// # }
+    /// ```
```

Comment:
So, after reading this I was immediately wondering if I could use bind multiple times. I thing this should be part of the docs.

And I know this will add a bunch of copypasta code but I'd split this up into _three_ examples: {non-postgres, postgres} ⨯ {single bind, multiple binds}. Maybe a `# PostgreSQL syntax` section with a sentence explaining that this is backend-specific and the two examples with `$n` instead of `?`.